src/transformStateWithClones.js

```diff
@@ -7,7 +7,33 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const statesArray = [];
+  const modifiedStates = { ...state };
+
+  for (const key of actions) {
+    const type = key.type;
+
+    switch (type) {
+      case 'addProperties':
+        Object.assign(modifiedStates, key.extraData);
+        statesArray.push({ ...modifiedStates });
+        break;
+      case 'removeProperties':
+        for (let x = 0; x < key.keysToRemove.length; x++) {
+          delete modifiedStates[key.keysToRemove[x]];
+        };
+        statesArray.push({ ...modifiedStates });
```

Comment:
No needs to use statesArray.push({ ...modifiedStates }); at each switch case. 
Just use it once (after! switch statement at the end of loop)